util/options_helper.h

```diff
@@ -625,6 +625,9 @@ static std::unordered_map<std::string, OptionTypeInfo>
         {"index_per_partition",
          {offsetof(struct BlockBasedTableOptions, index_per_partition),
           OptionType::kUInt64T, OptionVerificationType::kNormal, false, 0}},
+        {"partition_filters",
```

Comment:
First, in general they should not have two come together and the implementation should be improved by future patches (as we discussed offline) to separate them.

In the particular case that it is suggested here (index_type==kTwolevelIndexSearch and filter_policy != null) does not still mean that the filter has to be partitioned. The other way around is true though: if filter is partitioned index has to be partitioned too. But since this restriction is to be lifted in future, I think it is better to keep their configuration separately.